lib/pf/api.pm

```diff
@@ -1360,9 +1361,50 @@ sub handle_accounting_metadata : Public {
             pf::log::get_logger->debug("Not handling iplog update because we're not configured to do so on accounting packets.");
         }
     }
+    $client->notify("firewallsso_accounting", %RAD_REQUEST);
+
     return $return;
 }
 
+=head2 firewallsso_accounting
+
+Update the firewall sso based on radius accounting
+
+=cut
+
+sub firewallsso_accounting : Public {
+    my ($class, %RAD_REQUEST) = @_;
+    my $logger = pf::log::get_logger();
+    if ($RAD_REQUEST{'Calling-Station-Id'} && $RAD_REQUEST{'Framed-IP-Address'} && pf::util::isenabled($pf::config::Config{advanced}{sso_on_accounting})) {
+        my $mac = pf::util::clean_mac($RAD_REQUEST{'Calling-Station-Id'});
+        my $node = pf::node::node_attributes($mac);
+        my $ip = $RAD_REQUEST{'Framed-IP-Address'};
+        if($ip){
```

Comment:
Seems this can never be false as you're already checking it in the top condition

IMO you should split the top condition to output a meaningful warn message when not finding an IP or MAC so we can know what is up